### PR TITLE
Aswathy/Timeline

### PR DESCRIPTION
**Pull request by aswathy-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/conference/timeline/data_schedule.tsx [hunk 1]
```diff
@@ -0,0 +1,394 @@
+import React from 'react'
```
Comment (prince-deriv) [inline src/pages/conference/timeline/data_schedule.tsx:1]: there's no component rendered here, please remove the import and rename the file extension from `tsx` to `ts`

File: src/pages/conference/timeline/index.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    const onTabClick = (chosenTab: string) => {
+        setTab(chosenTab)
+    }
+
+    return (
+        <Layout type="conference" margin_top="0px">
+            <HeroWrapper />
+            <Tab>
+                <TabItem active={isDay1} onClick={() => onTabClick('dia1')}>
```
Comment (NikitK-deriv) [inline src/pages/conference/timeline/index.tsx:73]:
```suggestion
                <TabItem active={isDay1} onClick={onTabClick('dia1')}>
```
`onTabClick` is already callback

File: src/pages/conference/timeline/index.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                    <Header type="subtitle-1" align="center">
```
Comment (NikitK-deriv) [inline src/pages/conference/timeline/index.tsx:74]: Please use `as='div/h*'` property. By default Header is `h2`

File: src/pages/conference/timeline/index.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import styled, { css } from 'styled-components'
+import HeroBanner from './images/hero_banner_new.png'
+import ScheduleConference from './_schedule'
+import events_data from './_data-schedule'
+import { Header } from 'components/elements'
+import Layout from 'components/layout/layout'
+
+type TabProps = {
+    active?: boolean
+    mobile_padding?: string
```
Comment (NikitK-deriv) [inline src/pages/conference/timeline/index.tsx:11]: Are we use this property somewhere?

File: src/pages/conference/timeline/index.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    display: flex;
+    flex-direction: row;
+    justify-content: center;
+    align-items: center;
+    padding: 24px 40px;
+    height: 84px;
+    background: var(--color-white);
+    cursor: pointer;
+    ${(props) => (props.active ? selected : notSelected)}
+`
```
Comment (NikitK-deriv) [inline src/pages/conference/timeline/index.tsx:57]:
```suggestion
const TabItem = styled.div<{active?: string}>`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 24px 40px;
    height: 84px;
    background: var(--color-white);
    cursor: pointer;
    ${({active}) => (active ? selected : notSelected)}
`
```
If we remove `mobile_padding` from `TabProps` we can pass types from suggestion

File: src/pages/conference/timeline/index.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+type TabProps = {
+    active?: boolean
+    mobile_padding?: string
+}
+
+const HeroWrapper = styled.div`
+    background-image: url(${HeroBanner});
+    background-repeat: no-repeat;
+    background-size: 100% 100%;
+    min-height: 72rem;
```
Comment (NikitK-deriv) [inline src/pages/conference/timeline/index.tsx:18]: Let's change min-height property for mobile. Looks a little bit weird 🙏

File: src/pages/conference/timeline/_schedule.tsx [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+                                    </ImageContainer>
+                                    <ContentWrapper>
+                                        {item.speakers.map((item) => {
+                                            return (
+                                                <>
+                                                    <Header
+                                                        align="left"
+                                                        type="subtitle-1"
+                                                        weight="700"
+                                                        width="230px"
```
Comment (NikitK-deriv) [inline src/pages/conference/timeline/_schedule.tsx:141]: We exactly need to give these properties for Header?
Especially weight

File: src/pages/conference/timeline/_data-schedule.tsx [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+    {
+        id: 'day_01',
+        title: <Localize translate_text="Day1" />,
+        date: <Localize translate_text="November" />,
+        period: <Localize translate_text="Registrations 9:30 - 10:00" />,
+        data: [
+            {
+                timings: [
+                    {
+                        time: <Localize translate_text="10:00 - 10:20" />,
```
Comment (NikitK-deriv) [inline src/pages/conference/timeline/_data-schedule.tsx:14]: Are we need to localize a time?

File: src/pages/conference/timeline/_schedule.tsx [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+    }
+`
+
+const ScheduleConference = ({ item }: ConferenceProps) => {
+    const { id, date, data, period, title } = item
+
+    return (
+        <>
+            <HeaderTitle key={id}>
+                <Header weight="400" type="subtitle-1" align="center" pt="16px">
```
Comment (NikitK-deriv) [inline src/pages/conference/timeline/_schedule.tsx:91]: By default header have as="h2" for SEO optimization. It's bad approach to keep it here

File: src/pages/conference/timeline/_schedule.tsx [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+                                    </ImageContainer>
+                                    <ContentWrapper>
+                                        {item.speakers.map((item) => {
+                                            return (
+                                                <>
+                                                    <Header
+                                                        as="div"
+                                                        align="left"
+                                                        type="subtitle-1"
+                                                        weight="700"
```
Comment (NikitK-deriv) [inline src/pages/conference/timeline/_schedule.tsx:139]: By default Header have property `weight: bold` it's equal `weight="700"`

Comment (aswathy-deriv) [reply src/pages/conference/timeline/_schedule.tsx:139]: okay

File: src/pages/conference/timeline/_schedule.tsx (hunk 9 above)
Comment (aswathy-deriv) [reply src/pages/conference/timeline/_schedule.tsx:91]: okay

File: src/pages/conference/timeline/_schedule.tsx [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+                            <MainContainer>
+                                <WrapperContainer>
+                                    <TimeWrapper>
+                                        {item.timings.map((item) => {
+                                            return (
+                                                <Header
+                                                    as="div"
+                                                    type="heading-3"
+                                                    align="center"
+                                                    weight="700"
```
Comment (NikitK-deriv) [inline src/pages/conference/timeline/_schedule.tsx:115]: We don't need this due the default properties

File: src/pages/conference/timeline/_schedule.tsx [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+                                    </ImageContainer>
+                                    <ContentWrapper>
+                                        {item.speakers.map((item) => {
+                                            return (
+                                                <>
+                                                    <Header
+                                                        as="div"
+                                                        align="left"
+                                                        type="subtitle-1"
+                                                        weight="bold"
```
Comment (NikitK-deriv) [inline src/pages/conference/timeline/_schedule.tsx:139]: We don't need this due the default properties

File: src/pages/conference/timeline/_schedule.tsx [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+        margin-left: 40px;
+    }
+`
+
+const TimeWrapper = styled.div`
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+
+    h2:last-child {
```
Comment (NikitK-deriv) [inline src/pages/conference/timeline/_schedule.tsx:66]: Is this behavior works after change `<Header as="h2 />` by default on `<Header as="div" />` in current implementation. If not, let's remove it

File: src/pages/conference/timeline/_schedule.tsx [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+    }
+`
+
+const ScheduleConference = ({ item }: ConferenceProps) => {
+    const { id, date, data, period, title } = item
+
+    return (
+        <>
+            <HeaderTitle key={id}>
+                <Header as="div" weight="400" type="subtitle-1" align="center" pt="16px">
```
Comment (NikitK-deriv) [inline src/pages/conference/timeline/_schedule.tsx:91]: Let's create separate styled component with repeatable properties such:
`
font-weight: normal;
text-align: center;
`
And use him in whole component where we have the same properties

File: src/pages/conference/timeline/_schedule.tsx (hunk 13 above)
Comment (aswathy-deriv) [reply src/pages/conference/timeline/_schedule.tsx:66]: we need to change it back to h2 only

File: src/pages/conference/timeline/_schedule.tsx (hunk 11 above)
Comment (aswathy-deriv) [reply src/pages/conference/timeline/_schedule.tsx]: removed the weight property

File: src/pages/conference/timeline/_schedule.tsx (hunk 12 above)
Comment (aswathy-deriv) [reply src/pages/conference/timeline/_schedule.tsx]: okay

File: src/pages/conference/timeline/_schedule.tsx (hunk 14 above)
Comment (aswathy-deriv) [reply src/pages/conference/timeline/_schedule.tsx]: okay created a reusable component

File: src/pages/conference/timeline/_data-schedule.tsx (hunk 8 above)
Comment (prince-deriv) [reply src/pages/conference/timeline/_data-schedule.tsx:14]: Let's keep this for now, we don't know the format of time on other languages

File: src/pages/conference/timeline/index.tsx [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+const TabItem = styled.div<{ active?: boolean }>`
+    display: flex;
+    flex-direction: row;
+    justify-content: center;
+    align-items: center;
+    padding: 24px 40px;
+    height: 84px;
+    background: var(--color-white);
+    cursor: pointer;
+    ${(props) => (props.active ? selected : notSelected)}
```
Comment (prince-deriv) [inline src/pages/conference/timeline/index.tsx:57]:
```suggestion
    ${(props) => (props.active ? selected : unselected)}
```

**Review by prince-deriv — APPROVED** [review]

**Pull request merged**